NetTopologySuite.IO.SqlServer/SqlServerSpatialWriter.cs

```diff
@@ -0,0 +1,245 @@
+
+            using (var writer = new BinaryWriter(stream))
+            {
+                geography.WriteTo(writer);
+            }
+        }
+
+        private Geography ToGeography(IGeometry geometry)
+        {
+            if (geometry == null)
+            {
+                return new Geography { SRID = -1 };
+            }
+
+            var geometries = new Queue<(IGeometry, int)>();
+            geometries.Enqueue((geometry, -1));
+
+            // TODO: For geography (ellipsoidal) data, set IsLargerThanAHemisphere
+            var geography = new Geography
+            {
+                SRID = Math.Max(0, geometry.SRID),
+                IsValid = geometry.IsValid
+            };
+
+            while (geometries.Count > 0)
+            {
+                var (currentGeometry, parentOffset) = geometries.Dequeue();
+
+                var figureOffset = geography.Figures.Count;
+                var figureAdded = false;
+
+                switch (currentGeometry)
+                {
+                    case IPoint point:
+                    case ILineString lineString:
+                        figureAdded = addFigure(currentGeometry, FigureAttribute.Line);
+                        break;
+
+                    case IPolygon polygon:
+                        // NB: For geography (ellipsoidal) data, the actual shell is the ring oriented counter-clockwise
+                        figureAdded = addFigure(polygon.Shell, FigureAttribute.Line);
+                        foreach (var hole in polygon.Holes)
+                        {
+                            // NB: For geography (ellipsoidal) data, the actual holes are the rings oriented clockwise
+                            figureAdded |= addFigure(hole, FigureAttribute.Line);
+                        }
+                        break;
+
+                    case IGeometryCollection geometryCollection:
+                        foreach (var item in geometryCollection.Geometries)
+                        {
+                            geometries.Enqueue((item, geography.Shapes.Count));
+                            figureAdded = true;
+                        }
+                        break;
+
+                    default:
+                        throw new InvalidOperationException(
+                            string.Format(Resources.UnexpectedGeometryType, geometry.GetType()));
+                }
+
+                geography.Shapes.Add(
+                    new SqlShape
+                    {
+                        ParentOffset = parentOffset,
+                        FigureOffset = figureAdded ? figureOffset : -1,
+                        Type = ToOpenGisType(currentGeometry.OgcGeometryType)
+                    });
+
+                bool addFigure(IGeometry g, FigureAttribute figureAttribute)
+                {
+                    var pointOffset = geography.Points.Count;
+                    var pointsAdded = false;
+
+                    foreach (var coordinate in g.Coordinates)
```

Comment:
Still want me to use a filter?

Reply:
I don't think it's a blocker here.

Reply:
K, I still intend to do some basic profiling as part of aspnet/EntityFrameworkCore#13144